frontend/src/scenes/web-analytics/WebAnalyticsModal.tsx

```diff
@@ -6,17 +6,21 @@ import { LemonModal } from 'lib/lemon-ui/LemonModal'
 import { addProductIntentForCrossSell, ProductIntentContext } from 'lib/utils/product-intents'
 import { urls } from 'scenes/urls'
 import { WebQuery } from 'scenes/web-analytics/tiles/WebAnalyticsTile'
-import { webAnalyticsLogic } from 'scenes/web-analytics/webAnalyticsLogic'
-import { WebPropertyFilters } from 'scenes/web-analytics/WebPropertyFilters'
 
 import { ProductKey } from '~/types'
 
+import { webAnalyticsLogic } from './webAnalyticsLogic'
+import { webAnalyticsModalLogic } from './webAnalyticsModalLogic'
+import { WebPropertyFilters } from './WebPropertyFilters'
+
 export const WebAnalyticsModal = (): JSX.Element | null => {
     const {
-        modal,
         dateFilter: { dateFrom, dateTo },
     } = useValues(webAnalyticsLogic)
-    const { closeModal, setDates } = useActions(webAnalyticsLogic)
+    const { modal } = useValues(webAnalyticsModalLogic)
+
+    const { setDates } = useActions(webAnalyticsLogic)
+    const { closeModal } = useActions(webAnalyticsModalLogic)
```

Comment:
To guarantee you're connecting to the right logic - and the inner logic is connecting to the right `webAnalyticsLogic` - you should add a `BindLogic` to `WebAnalyticsDashboard`. This guarantees all child components connect to the same instance of the logic.

This isn't a *requirement* because the logic is unique now (their `path` + `props` tuple is constant) but it might be a good thing to do this now to avoid bugs in the future